### PR TITLE
feat: Adds Order Summary Component

### PR DESCRIPTION
**Pull request by danzanzini**
## What's the purpose of this pull request?
This PR aims to create the CartSummary component following this [jira task](https://vtex-dev.atlassian.net/browse/FSSS-135).

## How it works? 
The `cart/OrderSummary` component was created under the `cart` folder. 
All Cart-related props are retrieved in their parent `CartSidebar`. The checkout button is also generated at the parent to avoid sending a lot of props to the summer component. The button still lacks some behaviors that will be handled later.

## How to test it?
Go to the [preview page](https://preview-253--base.preview.vtex.app), add some items to the cart, then open it. The summary part should be looking like this:
image

### CODE REVIEW

File: src/components/cart/OrderSummary/OrderSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import type { InputHTMLAttributes, ReactNode } from 'react'
+import React from 'react'
+import { useFormattedPrice } from 'src/sdk/product/useFormattedPrice'
+
+import './order-summary.scss'
+
+export interface OrderSummaryProps extends InputHTMLAttributes<HTMLDivElement> {
```
Comment (filipewl) [inline src/components/cart/OrderSummary/OrderSummary.tsx:7]: This interface is extending from `InputHTMLAttributes<HTMLDivElement>`, but none of the other possible props are really being used by the component, so I suggest we keep it simple:

```suggestion
export interface OrderSummaryProps {
```

File: src/components/cart/CartSidebar/CartSidebar.tsx [hunk 2]
```diff
@@ -31,11 +32,22 @@ function CartSidebar() {
...
-      <Button {...btnProps}>{isValidating ? 'loading...' : 'Checkout'}</Button>
+      <OrderSummary
+        subTotal={subTotal}
+        total={total}
+        numberOfItems={totalItems}
+        checkoutButton={
+          <Button
+            data-cart-checkout-button
+            variant="primary"
+            icon={<ArrowRightIcon />}
```
Comment (hellofanny) [inline src/components/cart/CartSidebar/CartSidebar.tsx:43]:
```suggestion
            icon={<ArrowRightIcon size={18} />}
```

File: src/components/cart/OrderSummary/OrderSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  total,
+  numberOfItems,
+  checkoutButton,
+}: OrderSummaryProps) {
+  const discount = subTotal - total
+  const formattedDiscount = useFormattedPrice(discount)
+
+  return (
+    <div className="order-summary" data-order-summary>
+      <p data-order-summary-subtotal>
```
Comment (hellofanny) [inline src/components/cart/OrderSummary/OrderSummary.tsx:25]: Maybe we can use these items inside a `List` or `Table` component.

File: src/components/cart/OrderSummary/OrderSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  subTotal,
+  total,
+  numberOfItems,
+  checkoutButton,
+}: OrderSummaryProps) {
+  const discount = subTotal - total
+  const formattedDiscount = useFormattedPrice(discount)
+
+  return (
+    <div className="order-summary" data-order-summary>
```
Comment (hellofanny) [inline src/components/cart/OrderSummary/OrderSummary.tsx:24]:
```suggestion
    <footer className="order-summary" data-order-summary>
```

File: src/components/cart/OrderSummary/OrderSummary.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          <div>Discount</div>
+          <div>-{formattedDiscount}</div>
+        </p>
+      )}
+      <p className="title-subsection" data-order-summary-total>
+        <div>Total</div>
+        <div>{useFormattedPrice(total)}</div>
+      </p>
+      {checkoutButton}
+    </div>
```
Comment (hellofanny) [inline src/components/cart/OrderSummary/OrderSummary.tsx:40]:
```suggestion
    </footer>
```

File: src/components/cart/OrderSummary/OrderSummary.tsx (hunk 3 above)
Comment (hellofanny) [reply src/components/cart/OrderSummary/OrderSummary.tsx:25]: Something like this:

```
<UIList>
  <li>
     <p data-order-summary-subtotal> ... </p>
  </li>
   <li>
     <p data-order-summary-discount> ... </p>
  </li>
</UList>
```

What do you think?

File: src/components/cart/OrderSummary/order-summary.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
+@import "../../../styles/scaffold";
+
+.order-summary[data-order-summary] {
+  padding: 1rem;
```
Comment (hellofanny) [inline src/components/cart/OrderSummary/order-summary.scss:4]: We are using our css tokens variables to style. You can find it on `styles/globals.scss`

Comment (hellofanny) [inline src/components/cart/OrderSummary/order-summary.scss:4]:
```suggestion
  padding: var(--space-3);
```

File: src/components/cart/OrderSummary/OrderSummary.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+}: OrderSummaryProps) {
+  const discount = subTotal - total
+  const formattedDiscount = useFormattedPrice(discount)
+
+  return (
+    <div className="order-summary" data-order-summary>
+      <p data-order-summary-subtotal>
+        <div>Subtotal ({numberOfItems} products)</div>
+        <div>{useFormattedPrice(subTotal)}</div>
+      </p>
```
Comment (filipewl) [inline src/components/cart/OrderSummary/OrderSummary.tsx:28]: I wonder if we can use an unordered list here! Something like the following (suggestion not to be committed): 

```suggestion
    <ul className="order-summary" data-order-summary>
      <li data-order-summary-subtotal>
        <span>Subtotal ({numberOfItems} products)</span>
        <span>{useFormattedPrice(subTotal)}</span>
      </li>
```

---

This way we can also keep all the common CSS under the `li` styles and avoid the [current repetition](https://github.com/vtex-sites/base.store/blob/411b3b42f83f4aa5cc4d77410cc644fc57fb2a73/src/components/cart/OrderSummary/order-summary.scss#L6-L22).
```scss
li {
    display: flex;
    justify-content: space-between;
    line-height: 1.5;
	&[data-order-summary-discount] { color: green; }
}
```

File: src/components/cart/OrderSummary/order-summary.scss [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
```
Comment (hellofanny) [inline src/components/cart/OrderSummary/order-summary.scss:5]: Maybe we can simplify like this:

```
[data-order-summary-subtotal], 
[data-order-summary-discount],
[data-order-summary-total] {
     display: flex;
     justify-content: space-between;
     line-height: 1.5;
   }

   [data-order-summary-discount] {
     color: green;
   }

```
 or we could have data-attribute only for `[data-order-summary-discount]`
 
```
p {
     display: flex;
     justify-content: space-between;
     line-height: 1.5;
   }
 
    [data-order-summary-discount] {
     color: green;
   }
```


What do you think?

File: src/components/cart/OrderSummary/order-summary.scss [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  [data-order-summary-subtotal] {
+    display: flex;
+    justify-content: space-between;
+    line-height: 1.5;
+  }
+
+  [data-order-summary-discount] {
+    display: flex;
+    justify-content: space-between;
+    color: green;
```
Comment (filipewl) [inline src/components/cart/OrderSummary/order-summary.scss:15]: There should be some CSS custom property we can use here (e.g. `color: var(--color-text-discount)`). 

Do we already have that? cc @hellofanny and @renatamottam

**Review by filipewl — COMMENTED** [review]
Looking good Dan! Left some suggestions to think about.

**Review by hellofanny — CHANGES_REQUESTED** [review]
Nice!! 🚀 🚀 🤩 
I've left some comments 👍

File: src/components/cart/OrderSummary/OrderSummary.tsx (hunk 7 above)
Comment (filipewl) [reply src/components/cart/OrderSummary/OrderSummary.tsx:28]: Reminded me of PetPlate's Checkout Summary. 
🙊 

CleanShot 2022-01-18 at 18 30 39

cc @hellofanny 💙

Comment (hellofanny) [reply src/components/cart/OrderSummary/OrderSummary.tsx:28]: I suggested the same thing ahahah 💛  , But instead we can use Faststore's List component. What do you think?

File: src/components/cart/OrderSummary/order-summary.scss (hunk 9 above)
Comment (hellofanny) [reply src/components/cart/OrderSummary/order-summary.scss:15]: We don't! 

Dan, can you please add this new variable?
- on `globals.scss` file (on the // Contextual Colors section), please add `--color-success-dark: #368369`
- on `colors.scss` file  (on the   // Text section), please add `--color-text-success: var(--color-success-dark)`

And then, replace `green` for `var(--color-text-success)`

@renatamottam agree?

File: src/components/cart/OrderSummary/OrderSummary.tsx (hunk 4 above)
Comment (danzanzini) [reply src/components/cart/OrderSummary/OrderSummary.tsx]: I've used `<List`

File: src/components/cart/OrderSummary/OrderSummary.tsx (hunk 7 above)
Comment (danzanzini) [reply src/components/cart/OrderSummary/OrderSummary.tsx]: I've done this using the `List` component. Thanks <3

File: src/components/cart/OrderSummary/order-summary.scss (hunk 9 above)
Comment (danzanzini) [reply src/components/cart/OrderSummary/order-summary.scss]: Thanks, @hellofanny. It's done like you said :D

**Review by hellofanny — APPROVED** [review]

**Review by filipewl — APPROVED** [review]

**Review by eduardoformiga — COMMENTED** [review]
LGTM, Dan! 👏  
just left one suggestion regarding the scaffold import

**Pull request merged**